src/core/opamProcess.ml

```diff
@@ -450,6 +450,8 @@ let create ?info_file ?env_file ?(allow_stdin=not Sys.win32) ?stdout_file ?stder
       close_stdin  ();
       close_stdout ();
       close_stderr ();
+      begin try Unix.chdir oldcwd
+      with Unix.Unix_error _ -> () end;
```

Comment:
Looks obviously correct but couldn’t we merge both uses of `Unix.chdir` using `Fun.protect` starting below the first `Unix.chdir` at the top of the function? This way if another reraise or uncaught exception occures it will be handled automatically

Reply:
Yeah, I should having avoided being lazy (the laziness is having to add `Fun.protect` to `OpamCompat` <sigh>)